core/helpers/causeDebt.ts

```diff
@@ -0,0 +1,118 @@
+import getContract, { getContractAddressByName } from '../src/contracts';
+import BigNumber from '../src/bignumber';
+import { WAD_NUMBER_OF_DIGITS } from '../src/constants/UNITS';
+import { EthereumProvider } from 'hardhat/types';
+import { ethers } from 'ethers';
+import { formatToHex } from './format';
+
+export const causeDebt = async (
+    network: string,
+    provider: EthereumProvider,
+    debtAmountDai: BigNumber,
+    mkrOnAuction: BigNumber,
+    daiOnAuction: BigNumber
+) => {
+    await overwriteQueuedDebt(network, provider, debtAmountDai);
+    await ensureQueuedDebtEqual(network, debtAmountDai);
+    await overwriteProtocolOwnDaiBalance(network, provider);
+    await ensureDaiIsZero(network);
+    await overwriteLottedMkrAmount(network, provider, mkrOnAuction);
+    await ensureLottedMkrHasValue(network, mkrOnAuction);
```

Comment:
Here as well. What is `LottedMkrAmount`?

Reply:
well, this one is quite literal.

https://www.merriam-webster.com/dictionary/lotted

Reply:
We've used `receiveAmountMKR`, `desiredMkrAmount` and called similar things `compensation` on the frontend so far.

Although in order to suggest something, I need to understand first what is it. What is `LottedMkrAmount` and why do you need to overwrite it?

Reply:
new name added, pls see.